DataManager/src/DataManagement/RequestManager.java

```diff
@@ -41,19 +40,39 @@
 	@SuppressWarnings ( { "unchecked", "rawtypes" } )
 	private static List<Object> getSubMenu ( String langName )
 	{
-		List<Object> returnList = new ArrayList();
-		String[][] array = database.request( langName );
+		boolean langAvail = false;
+		String[] tmp = database.request();
 		
-		for ( int i = 0; i < array.length; i++ )
+		for ( String i : tmp )
 		{
-			if ( !array[i][0].equals( langName ) )
+			if ( i.equalsIgnoreCase( langName ) )
 			{
-				returnList.add( new Language( array[i][0], Integer.parseInt( array[i][1] ) ) );
+				langAvail = true;
+				break;
 			}
 		}
 		
-		Collections.sort( (List<Language>) (Object) returnList );
-		returnList.add( 0, "sub_menu_list" );
+		List<Object> returnList = new ArrayList();
+		
+		if ( langAvail )
+		{
+			String[][] array = database.request( langName );
+			
+			for ( int i = 0; i < array.length; i++ )
+			{
+				if ( !array[i][0].equals( langName ) )
+				{
+					returnList.add( new Language( array[i][0], Integer.parseInt( array[i][1] ) ) );
+				}
+			}
+			
+			Collections.sort( (List<Language>) (Object) returnList );
+			returnList.add( 0, "sub_menu_list" );
+		}
+		else
+		{
+			returnList.add( "Language Unavailable" );
+		}
 		
```

Comment:
There is only 1 loop here, everything else is a check to ensure that the request is valid and a correct response can be generated.